tests/test_parameters.py

```diff
@@ -587,3 +587,17 @@ def test_compare_key_value_list():
 )
 def test_parameter_can_be_pickled(p):
     pickle.dumps(p)
+
+
+def test_add_remove_param_setup_success():
+    """Checks that add_param and remove_param don't require m.setup() to be run prior to constructing a fitter"""
+    m = get_model(os.path.join(datadir, "B1855+09_NANOGrav_9yv1.gls.par"))
+    t = get_TOAs(os.path.join(datadir, "B1855+09_NANOGrav_9yv1.tim"))
+    FD4 = prefixParameter(parameter_type="float", name="FD4", value=0.0, units=u.s)
+    m.add_param_from_top(FD4, "FD")
+    assert len(m.components["FD"].params) == 4
+    """Fitter construction used to fail after remove_param without m.setup(). Test this:"""
```

Comment:
This would be good detail to put in the test title!

Reply:
I'm just hoping the test title can tell me what property it's testing - `test_add_param_increases_number_of_params`, `test_fitter_can_be_constructed_after_remove_param` - so it's more obvious what feature the test is testing. I realize the test currently checks both of these, but it would be better if it could be two tests so we'd get separate answers on whether those two things worked. If the test is checking for a specific bug, it can make sense to include the bug number in the test name. (Nobody ever calls tests by hand so it's not particularly a problem if their names are long.)

My reasoning is, suppose I ran `make test` and your test here failed. What would I know went wrong? What was supposed to happen but isn't? How well does it help me find the place I broke something?

Reply:
I'm going to modify what's now the second test again, so not quite ready for review.